### PR TITLE
feat(editor): add find & replace support in editor

### LINKED ISSUE #129: **Find & Replace** inside an editor — find bar already exists (AppKit `usesFindBar`); replace UI is non-trivial.
## Summary
Add replace support to the editor find experience. The basic AppKit find bar already exists through `usesFindBar`, but Alas still needs an integrated replace flow that works inside editor panes without fighting the native find UI.

## Desired behavior
- Users can find text inside an editor pane as they do today.
- Users can replace the current match.
- Users can replace all matches in the current editor document.
- Replacement should apply to the editor buffer and preserve the normal dirty/save behavior.
- The UI should stay scoped to the active editor pane, not global app search.

## Acceptance criteria
- Find continues to work with the existing find bar behavior.
- Replace-one and replace-all are available from the editor find UI or an adjacent, discoverable editor UI.
- Replacements update the active document contents and can be undone through normal editor undo.
- Replace-all reports or clearly reflects the number of replacements where practical.
- Works for common plain-text cases, including repeated matches, no matches, and empty replacement text.

## Implementation notes
- AppKit `usesFindBar` gives native find UI but not a complete replace UI, so this may need a small custom replace control layered around the existing editor component.
- Keep selection/focus behavior predictable: after replacing one match, move to the next match where possible.
- Avoid broad project-wide search/replace in this task; this is only inside one editor document.

### PR DESCRIPTION
**Pull request by mrmans0n**
Implements GitHub issue #129: Find & Replace inside an editor.

## Changes

- **EditorFindController** — Plain-text find/replace controller scoped to a single `CodeTextView`. Uses `NSTextView.insertText(_:replacementRange:)` so undo, dirty tracking, highlighting, and LSP `didChange` flow through the normal pipeline. `replaceAll` wraps edits in a single undo grouping.

- **EditorFindBarView** — SwiftUI bar with find field, replace field, Replace/All/Done buttons, status message, and Enter key handling (`onSubmit` triggers Find Next / Replace Current).

- **CodeTextView** — Added `autoPairDisabled` guard: when set, `insertText` bypasses auto-pair logic, preventing single-character replacements from inserting paired delimiters.

- **EditorTabView** — Conditionally renders `EditorFindBarView`. Listens for `.codeEditorDidAttach`/`.codeEditorDidDetach` to wire the controller to the active `CodeTextView`. Wired `onFind` for directional navigation.

- **CodeEditorCoordinator** — Posts `.codeEditorDidAttach`/`.codeEditorDidDetach` notifications. Removed unused `detachedTabId` variable. Fixed `tabId as Any` coercion warning.

- **AlasApp** — Added **Edit > Find and Replace** (⌘⌥F) menu command, disabled when no editor tab is active.

- **RootView** — Added `Notification.Name` extensions for `.alasShowFindReplace`, `.codeEditorDidAttach`, `.codeEditorDidDetach`.

- **Tests** — 10 focused `EditorFindControllerTests` covering replace current, replace all, empty cases, match counting, and navigation.

## Review fixes applied

- `autoPairDisabled` now actually bypasses auto-pair in `CodeTextView.insertText` (was set but never checked)
- `replaceCurrent()` advances forward to next match instead of backward
- Stale `text.length` replaced with fresh document length after mutation
- Enter key handlers added to find/replace text fields
- Unused `detachedTabId` removed from `CodeEditorCoordinator.detach()`

Fixes #129

### CODE REVIEW

File: Alas/Sources/Center/EditorTabView.swift [hunk 1]
```diff
@@ -42,6 +104,35 @@ struct EditorTabView: View {
...
+                  notifTabId == tabId else { return }
+            activeTextView = nil
+            findController.textView = nil
+        }
+        .onReceive(NotificationCenter.default.publisher(for: .alasShowFindReplace)) { _ in
+            guard appState.tabs.activeTabId(forWorktree: worktreeId) == tabId else { return }
+            findBarVisible = true
+            DispatchQueue.main.async {
+                if let textView = activeTextView {
+                    textView.window?.makeFirstResponder(textView)
```
[the remark this answers is not in the document]
Comment (mrmans0n) [reply Alas/Sources/Center/EditorTabView.swift]: Fixed in a9fa7da — find TextField now receives focus via @FocusState when the find bar opens.

File: Alas/Sources/Code/Editor/EditorFindController.swift [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        guard !findString.isEmpty else { return false }
+
+        let text = textView.string as NSString
+        var foundRange = textView.selectedRange()
+
+        // If the selection is not a valid match, search from the beginning.
+        let selectionIsMatch = foundRange.length == findString.utf16.count
+            && text.substring(with: foundRange) == findString
+        if !selectionIsMatch {
+            if let match = nextMatchRange(startingAt: 0) {
```
[the remark this answers is not in the document]
Comment (mrmans0n) [reply Alas/Sources/Code/Editor/EditorFindController.swift]: Fixed in a9fa7da — replaceCurrent now starts from the cursor position and wraps to offset 0 when no match is found ahead.

File: Alas/Sources/Code/Editor/EditorFindController.swift [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            }
+        }
+
+        guard foundRange.length == findString.utf16.count,
+              text.substring(with: foundRange) == findString else { return false }
+
+        textView.autoPairDisabled = true
+        defer { textView.autoPairDisabled = false }
+
+        textView.insertText(replacementString, replacementRange: foundRange)
```
[the remark this answers is not in the document]
Comment (mrmans0n) [reply Alas/Sources/Code/Editor/EditorFindController.swift:81]: Fixed in a9fa7da — both replaceCurrent and replaceAll now guard on textView.isEditable, returning false/0 on read-only editors.
